packages/grafana-ui/src/components/Select/SelectMenu.tsx

```diff
@@ -57,7 +58,7 @@ export const VirtualizedSelectMenu = ({ children, maxHeight, options, getValue }
   }
 
   const longestOption = max(options.map((option) => option.label?.length)) ?? 0;
-  const widthEstimate = longestOption * VIRTUAL_LIST_WIDTH_ESTIMATE_MULTIPLIER;
+  const widthEstimate = longestOption * VIRTUAL_LIST_WIDTH_ESTIMATE_MULTIPLIER + VIRTUAL_LIST_PADDING * 2;
```

Comment:
That is cool! I'm assuming if its not great performance we wouldn't want to run `measureText` on thousands of options though... so would we first find the one with longest amount of characters and then measure that one? In the example on storybook there are thousands of options that have the same number of characters (11), from "Option 1000" to "Option 9999", but then they all would measure differently depending on which one is picked. I have tested it with picking the "first one" (Option 1000) and it looks like this:

<img width="119" alt="Screenshot 2024-04-17 at 16 19 53" src="https://github.com/grafana/grafana/assets/100691367/1acd998b-6761-4fe6-bfc5-cab4b51e1f38">
<img width="123" alt="Screenshot 2024-04-17 at 16 20 02" src="https://github.com/grafana/grafana/assets/100691367/32a15467-1bfc-456b-b539-90bfe66977e9">

You can see how for Option 9999 is a bit tighter. Any ideas?

Reply:
Oh yes - we would only run it on the longest option in the list.

I figured it could create a more-accurate estimation for the width to use, but you're right that this is is a scenario where it works less great.

I take it in this case then the 8px `VIRTUAL_LIST_WIDTH_ESTIMATE_MULTIPLIER` represents a wider-than-average character? 

A different option might be to base this on the `ch` unit, which is the width of the `0` glyph:

> What I’ve found through random experimentation is that in proportional typefaces, `1ch` is usually wider than the average character width, usually by around 20-30%.  But there are at least a few typefaces where the zero symbol is skinny with respect to the other letterforms; in such a case, 1ch is narrower than the average character width.

https://meyerweb.com/eric/thoughts/2018/06/28/what-is-the-css-ch-unit/

I don't think there's anything terribly bad about the current solution (I figure we all recognise it's not ideal) - just want to float different options if you're interested in them.

Reply:
Both options I tried previously work dubiously for long values.
With measure text it doesnt show the whole text (it doesnt seem to measure it completely correct):
<img width="821" alt="Screenshot 2024-04-17 at 16 51 00" src="https://github.com/grafana/grafana/assets/100691367/7a49b1d1-275e-4e9e-ae4f-2b92116d42ce">

With the `VIRTUAL_LIST_WIDTH_ESTIMATE_MULTIPLIER` it adds too much space, like a lot:
<img width="989" alt="Screenshot 2024-04-17 at 16 51 53" src="https://github.com/grafana/grafana/assets/100691367/d317b4b9-8bf1-4b5a-8044-fd27364b3e27">

Assuming more space is better 🤷 
I think the current solution solves the problem even if not perfectly, so maybe we just merge it as is and then we can keep this in mind when we are building the new select? 😄

Reply:
Completely reasonable!